src/AutoUI/index.tsx

```diff
@@ -227,28 +259,43 @@ export const AutoUI = <T extends AutoUIBaseResource<T>>({
 				internalPagination.itemsPerPage,
 			);
 		},
-		[setFilters, onChange],
+		[
+			setFilters,
+			internalOnChange,
+			internalPagination.itemsPerPage,
+			internalPagination.page,
+			sort,
+		],
 	);
 
 	const $setSelected = React.useCallback<
 		CollectionLensRendererProps<T>['changeSelected']
 	>(
-		(items, checkedState = undefined) => {
+		(items, newCheckedState = undefined) => {
 			setSelected(items);
-			setCheckedState(checkedState ?? 'none');
-			if (actionData) {
-				setActionData({ ...actionData, affectedEntries: items, checkedState });
-			}
+			setCheckedState(newCheckedState ?? 'none');
+			setActionData((oldState) =>
+				oldState
+					? {
+							...oldState,
+							affectedEntries: items,
+							checkedState: newCheckedState,
+						}
+					: undefined,
+			);
 		},
-		[setSelected, setActionData, actionData],
+		[setSelected, setActionData],
 	);
+
+	const serverSide = pagination?.serverSide;
+	const totalItems = serverSide ? pagination.totalItems : undefined;
```

Comment:
Once we're away from rendition we should update the `Pagination` typing from
```
export type Pagination = {
    itemsPerPage?: number;
} & ({
    serverSide?: undefined;
} | {
    serverSide: true;
    currentPage: number;
    totalItems: number;
});
```
to
```
export type Pagination = {
    itemsPerPage?: number;
} | {
    currentPage: number;
    totalItems: number;
});
```
(the latter of the two typings would be for serverside)

I say this based on how I see us using it in the UI code, so if I'm mistaken or missing something, let me know